app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/deployManifestConfig.html

```diff
@@ -1,11 +1,44 @@
-<div ng-if="ctrl.state.loaded">
+<div ng-if="ctrl.state.loaded" xmlns="http://www.w3.org/1999/html">
   <div class="alert alert-warning">
     <strong>Warning!</strong> This stage is under active development and is subject to change.
   </div>
-  <v2-wizard-page key="basicSettings" label="Basic Settings" mark-complete-on-view="false">
-    <kubernetes-manifest-basic-settings command="ctrl.$scope.stage" metadata="ctrl.metadata"> </kubernetes-manifest-basic-settings>
+
+  <v2-wizard-page key="basicSettings" label="Basic Settings">
+    <kubernetes-manifest-basic-settings command="ctrl.$scope.stage"
+                                        metadata="ctrl.metadata">
+    </kubernetes-manifest-basic-settings>
   </v2-wizard-page>
-  <v2-wizard-page key="manifest" label="Manifest" mark-complete-on-view="false">
-    <kubernetes-manifest-entry command="ctrl.$scope.stage" metadata="ctrl.metadata" change="ctrl.change()"> </kubernetes-manifest-entry>
+
+  <v2-wizard-page key="manifest" label="Manifest">
+    <div class="container-fluid form-horizontal">
+      <ng-form name="kubernetesManifestSource">
+        <div class="form-group form-inline">
+          <div class="col-md-3 sm-label-right">
+            Manifest Source
+            <help-field key="kubernetes.manifest.source"></help-field>
+          </div>
+          <div class="col-md-4">
+            <select class="form-control input-sm" ng-model="ctrl.$scope.stage.source">
```

Comment:
If there are a small number of these (< 4), I think it's a better UI pattern to use radio buttons.

Reply:
Oh good call